src/structure/pdb.jl

```diff
@@ -30,26 +49,326 @@ end
+                download("http://files.rcsb.org/download/$pdbid"*pdbextension[file_format]*"$ba_number.gz",archivefilepath)
+            elseif file_format == mmCIF
+                download("http://files.rcsb.org/download/$pdbid-assembly$ba_number"*pdbextension[file_format]*".gz", archivefilepath)
+            else
+                warn("Biological Assembly is available only in PDB and mmCIF formats!")
+            end
+        end
+        # Verify if the compressed PDB file is downloaded properly and extract it. For MMTF no extraction is needed
+        if ispath(archivefilepath) && filesize(archivefilepath) > 0 && file_format != MMTF           
+            input = open(archivefilepath) |> ZlibInflateInputStream
+            open(pdbpath,"w") do output
+                for line in eachline(input, chomp=false)
+                    print(output, line)
+                end
+            end
+            close(input)
+        end
+        # Verify if the PDB file is downloaded and extracted without any error
+        if !ispath(pdbpath) || filesize(pdbpath)==0
+            throw(ErrorException("Error downloading PDB : $pdbid"))
+        end
+        # Remove the temporary compressd PDB file downloaded to clear up space
+        if ispath(archivefilepath)
+            rm(archivefilepath)
+        end
     end
 end
 
 
+"""
+Downloads a list of PDB files from the RCSB PDB server. 
+By default downloads the PDB files to current working directory; 
+if the keyword argument `pdb_dir` is set the PDB files are downloaded to the
+specified directory;
+if the keyword argument `obsolete` is set `true`, the PDB files are downloaded
+into the obsolete directory inside `pdb_dir`;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the
+PDB file if it exists in the `pdb_dir`.
+"""
+function downloadpdb(pdbidlist::Array{String,1}; kwargs...)
+    failedlist = String[]
+    for pdbid in pdbidlist      
+        try
+            downloadpdb(pdbid; kwargs...)
+        catch
+            warn("Error downloading PDB : $pdbid")
+            push!(failedlist,pdbid)
+        end
+    end
+    if length(failedlist) > 0
+        warn(length(failedlist)," PDB files failed to download : ", failedlist)
+    end
+end
+
+
+"""
+Downloads the entire PDB files available in the RCSB PDB server. 
+By default downloads the PDB files to current working directory; 
+if the keyword argument `pdb_dir` is set the PDB files are downloaded to the
+specified directory;
+if the keyword argument `overwrite` is set `true`, then it will overwrite the
+PDB file if exists in the `pdb_dir`.
+"""
+function downloadentirepdb(;pdb_dir::AbstractString=pwd(), file_format::Type=PDB, overwrite::Bool=false)
+    # Get the list of all pdb entries from RCSB PDB Server using getallpdbentries() and downloads them
+    pdblist = pdbentrylist()
+    info("About to download "*string(length(pdblist))*" PDB files. Make sure to have enough disk space and time!")
```

Comment:
Why don't you use string interpolation? `"... download $(length(pdblist)) PDB files..."`